src/core/src/layout.cpp

```diff
@@ -640,7 +640,11 @@ void AttributeAdapter<ov::Layout>::set(const std::string& value) {
 bool LayoutAttribute::visit_attributes(AttributeVisitor& visitor) {
     std::string layout_str = value.to_string();
     visitor.on_attribute("layout", layout_str);
-    value = Layout(layout_str);
+    // some attribute visitor will not change the value
+    // for example, rt info serializer
+    // in this case, parallelization can be supported in hash pass
+    if (layout_str != value.to_string())
```

Comment:
How this check fix race condition? Can you explain?
I think it only reduce chance as number of writes to `value` will reduced.

Reply:
Hi @praasz , generally visitors are able to change the attribute values and callers are responsible for the thread-safe.
while in this bug, the visitor is rt info serializer from ov::Hash pass, which will only read the value, so the fix here is to avoid unnecessary change of value. otherwise, we have to add global mutex in icore before do hashing.

Reply:
The presented solution can decrease the reproduced probability but cannot avoid race-condition completely. Considered the original probability is very low, I likely don't think we have enough motivation to add extra global mutex.

Reply:
As you mention this solution reduce probability not fix. Consider create follow-up ticket to fix root cause (not necessary by using mutex).
It looks like issue is when visiting attribute with same value it want change it but will write (this fix reduce such writes). If `ov::Hash` just reads attribute then maybe visitor should only access it for reading and not change it e.g usign visitor like const object, but when visitor could change value use it as non-const. But it requires more investigate and I don't have solution now.